ext/standard/proc_open.c

```diff
@@ -932,25 +1009,8 @@ PHP_FUNCTION(proc_open)
 		descriptors[ndesc].index = (int)nindex;
 
 		if (Z_TYPE_P(descitem) == IS_RESOURCE) {
-			/* should be a stream - try and dup the descriptor */
-			php_stream *stream = (php_stream*)zend_fetch_resource(Z_RES_P(descitem), "stream", php_file_le_stream());
-			if (stream == NULL) {
-				goto exit_fail;
-			}
-
-			php_socket_t fd;
-			php_file_descriptor_t desc;
-
-			if (FAILURE == php_stream_cast(stream, PHP_STREAM_AS_FD, (void **)&fd, REPORT_ERRORS)) {
-				goto exit_fail;
-			}
-
-#ifdef PHP_WIN32
-			desc = (HANDLE)_get_osfhandle(fd);
-#else
-			desc = fd;
-#endif
-			if (dup_proc_descriptor(desc, &descriptors[ndesc].childend, nindex) == FAILURE) {
+			if (set_proc_descriptor_from_resource(descitem, &descriptors[ndesc],
+				                                    ndesc) == FAILURE) {
```

Comment:
No argument alignment please. Either indent the next line or keep the line longer.

Reply:
OK. I was imitating your coding style there. (From the call to `set_proc_descriptor_from_array`.)

Will update both calls.